lib/services/platform-environment-requirements.ts

```diff
@@ -0,0 +1,135 @@
+		private $staticConfig: IStaticConfig) { }
+
+	public static CLOUD_BUILDS_OPTION_NAME = "Configure for Cloud Builds";
+	public static SETUP_SCRIPT_OPTION_NAME = "Configure for Local Builds";
+	public static MANUALLY_SETUP_OPTION_NAME = "Skip Step and Configure Manually";
+	private static BOTH_CLOUD_BUILDS_AND_SETUP_SCRIPT_OPTION_NAME = "Configure for Both Local and Cloud Builds";
+	private static NOT_CONFIGURED_ENV_MESSAGE = "To continue, choose one of the following options: ";
+	private static NOT_CONFIGURED_ENV_AFTER_SETUP_SCRIPT_MESSAGE = "The setup script was not able to configure your environment for local builds. To execute local builds, you have to set up your environment manually. To continue, choose one of the following options:";
+	private static NATIVESCRIPT_CLOUD_EXTENSION_NAME = "nativescript-cloud";
+
+	private cliCommandToCloudCommandName: IStringDictionary = {
+		"build": "tns cloud build",
+		"run": "tns cloud run",
+		"deploy": "tns cloud deploy"
+	};
+
+	public async checkEnvironmentRequirements(platform: string): Promise<boolean> {
+		const canExecute = await this.$doctorService.canExecuteLocalBuild(platform);
+		if (!canExecute) {
+			if (!isInteractive()) {
+				this.fail("You are missing the nativescript-cloud extension and you will not be able to execute cloud builds. Your environment is not configured properly and you will not be able to execute local builds. To continue, choose one of the following options: " + EOL
+					+ "Run $ tns setup command to run the setup script to try to automatically configure your environment for local builds." + EOL
+					+ "Run $ tns cloud setup command to install the nativescript-cloud extension to configure your environment for cloud builds." + EOL
+					+ `Verify that your environment is configured according to the system requirements described at ${this.$staticConfig.SYS_REQUIREMENTS_LINK}`);
+			}
+
+			this.$logger.info("You are missing the nativescript-cloud extension and you will not be able to execute cloud builds. Your environment is not configured properly and you will not be able to execute local builds. " + EOL
+				+ "Select “Configure for Cloud Builds” to install the nativescript-cloud extension and automatically configure your environment for cloud builds." + EOL
+				+ "Select “Configure for Local Builds” to run the setup script and automatically configure your environment for local builds."
+				+ "Select “Configure for Both Local and Cloud Builds” to automatically configure your environment for both options."
+				+ "Select “Skip Step and Configure Manually” to disregard these options and install any required components manually.");
+
+			const selectedOption = await this.$prompter.promptForChoice(PlatformEnvironmentRequirements.NOT_CONFIGURED_ENV_MESSAGE, [
+				PlatformEnvironmentRequirements.CLOUD_BUILDS_OPTION_NAME,
+				PlatformEnvironmentRequirements.SETUP_SCRIPT_OPTION_NAME,
+				PlatformEnvironmentRequirements.MANUALLY_SETUP_OPTION_NAME,
```

Comment:
I think it more logical the manual choice be the last option. My advice would be
1. Cloud
2. Local
3. Both
4. Manual

e.g. from easiest (most likely to succeed) to hardest (least likely to suceed). A personal preference of course - not a merge-stopper